### ISSUE TITLE
[BUG] Microsoft.Azure.Services.AppAuthentication + Visual Studio 16.2 Doesn't Work

### ISSUE THREAD

**Issue by wjchristenson2** [Azure/azure-sdk-for-net#7133]
Upgrading Visual Studio 2019 to 16.2.0 breaks Microsoft.Azure.Services.AppAuthentication regardless of AppAuthentication version. Tested on a couple of different machines and different developer logins. Developers are no longer able to use their VS credentials to authenticate to AKV after the upgrade. We've tried logging out of VS and back in, restarting VS, restarting machines to no avail. We were previously on 16.1.x.

> Parameters: Connection String: [No connection string specified], Resource: https://vault.azure.net, Authority: https://login.windows.net/c191f8c9-b4cb-47de-971b-dc01ee887392. Exception Message: Tried the following 4 methods to get an access token, but none of them worked. Parameters: Connection String: [No connection string specified], Resource: https://vault.azure.net, Authority: https://login.windows.net/c191f8c9-b4cb-47de-971b-dc01ee887392. Exception Message: Tried to get token using Managed Service Identity. Failed after 5 retries. Unable to connect to the Managed Service Identity (MSI) endpoint. Please check that you are running on an Azure resource that has MSI setup.
Parameters: Connection String: [No connection string specified], Resource: https://vault.azure.net, Authority: https://login.windows.net/c191f8c9-b4cb-47de-971b-dc01ee887392. Exception Message: Tried to get token using Visual Studio. Access token could not be acquired. Parameters: Connection String: [No connection string specified], Resource: https://vault.azure.net, Authority: https://login.windows.net/c191f8c9-b4cb-47de-971b-dc01ee887392. Exception Message: Tried to get token using Azure CLI. Access token could not be acquired. 'az' is not recognized as an internal or external command,
operable program or batch file. Parameters: Connection String: [No connection string specified], Resource: https://vault.azure.net, Authority: https://login.windows.net/c191f8c9-b4cb-47de-971b-dc01ee887392. Exception Message: Tried to get token using Active Directory Integrated Authentication. Access token could not be acquired. password_required_for_managed_user: Password is required for managed user

**Comment by wjchristenson2** [comment]
For others, we used Azure CLI to login from the command prompt as a work around.

Download: https://docs.microsoft.com/en-us/cli/azure/install-azure-cli?view=azure-cli-latest
Sign In (note you may have to select your subscription if you have access to more than 1 after you sign in): https://docs.microsoft.com/en-us/cli/azure/authenticate-azure-cli?view=azure-cli-latest

**Comment by AlexGhiondea** [comment]
@wjchristenson2 are you saying that logging in via the CLI addressed the issue? Or that logging in via the CLI is a workaround for people that can't connect to AKV using their VS credentials?

**Comment by wjchristenson2** [comment]
@AlexGhiondea logging in via the CLI is a workaround for people that can't connect to AKV using their VS credentials after upgrading to 16.2.0.

**Comment by crmann1** [comment]
In 16.2 can you try going to  Tools->options->Azure Service Authentication  and ensure the correct account is listed in the dialog, then click OK and try again.

**Comment by wjchristenson2** [comment]
@crmann1 Correct account was listed/selected in the dialog. 

Update: Yesterday we were able to connect to AKV using VS credentials. I logged out of CLI and I could still connect. I was also prompted to upgrade to VS 16.2.1 which I took. I could connect to AKV using VS credentials after 16.2.1 upgrade as well. I then uninstalled Azure CLI and I could still connect using VS credentials.

I had another developer take the update today. 16.2.0 was not an option so he took 16.2.1. He could connect to AKV using VS credentials after the upgrade.

We had 2 developers originally with the same issue when upgrading to 16.2.0. The other developer left on vacation so I can't troubleshoot further using his machine.

I'm going to close the issue. It seems the jump to 16.2.1 is working. Could have been a temp issue with 16.2.0.
